src/gtk/toga_gtk/widgets/internal/buttons/refresh.py

```diff
@@ -0,0 +1,190 @@
+from toga_gtk.libs import Gtk
+from .base import ParentPosition
+
+
+class RefreshButtonWidget(Gtk.Revealer):
+    def __init__(self, on_refresh: callable, on_close: callable, 
+                 position: Gtk.Align, margin: int, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.parent = None
+
+        self._refresh_btn = Gtk.Button.new_from_icon_name(
+            "view-refresh-symbolic", Gtk.IconSize.BUTTON)
+        self._refresh_btn.set_can_focus(False)
+        refresh_btn_context = self._refresh_btn.get_style_context()
+        refresh_btn_context.add_class("osd")
+        refresh_btn_context.add_class("toga-detailed-list-floating-buttons")
+
+        self._refresh_btn_handler = self._refresh_btn.connect(
+            "clicked",
+            lambda w: on_refresh())
+
+        self._close_btn = Gtk.Button.new_from_icon_name(
+            "close-symbolic", Gtk.IconSize.BUTTON)
+        self._close_btn.set_can_focus(False)
+        close_btn_context = self._close_btn.get_style_context()
+        close_btn_context.add_class("osd")
+        close_btn_context.add_class("toga-detailed-list-floating-buttons")
+        refresh_btn_context.remove_class("button")
```

Comment:
Is there a reason this is here, and not a few lines further up with all the other refresh button code?